apps/yggdrasil/priv/repo/migrations/20160229163000_create_role_permission.exs

```diff
@@ -0,0 +1,16 @@
+defmodule Yggdrasil.Repo.Migrations.CreateRoleResource do
+  use Ecto.Migration
+
+  def change do
+    create table(:roles_permissions, primary_key: false) do
+      add :role_id, references(:roles, on_delete: :nothing)
+      add :resource, references(:resources, on_delete: :nothing, column: :name, type: :string)
+      add :permission, references(:permissions, on_delete: :nothing, column: :name, type: :string)
+
+      timestamps
+    end
+    create index(:roles_permissions, [:role_id])
```

Comment:
I will need to read up on this for postgress. I don't know enough about indexes, but the specific thing that comes to mind as a reaction to your question is that if you are looking up by :role_id only does the unique constraint cover that? 

I don't know exactly how the indexes work if you only use one column all the ones defined in the index and in this case the primary lookup will be role_id only. Put another way will it use the index if you don't supply all columns?